core/Settings/Setting.php

```diff
@@ -158,6 +158,16 @@ public function configureField()
 
             $this->setUiControlIfNeeded($this->config);
             $this->checkType($this->config);
+            if (empty($this->config->availableValues) && in_array($this->type, array(FieldConfig::TYPE_FLOAT, FieldConfig::TYPE_INT, FieldConfig::TYPE_STRING))) {
```

Comment:
We might have to remove this again, or implement the HTML5 validation features into our settings to highlight the errors etc. Will need to be tested with MaterializeCSS etc. If this worked it would be epic.